addon/components/ownership-selection-modal.js

```diff
@@ -15,19 +19,42 @@ export default Component.extend({
   header: 'Share with',
   cta: 'Share',
 
+  items: computed('searchTerm', function() {
+    let members;
+
+    return this.currentUser.fetchColleagues().then((res) => {
+      members = res.users.filter((user) => { 
+        return user.active;
+      });
+
+      return members.filter((user) => {
+        const { first_name, last_name, email } = user;
+        return email.includes(this.searchTerm) ||
+        (first_name || '').includes(this.searchTerm) ||
+        (last_name || '').includes(this.searchTerm);
+      });
+    });
+  }),
+
   actions: {
+    searchEntities(text) {
+      this.set('searchTerm', text);
+    },
+
     updateOwnership() {
-      this.get('ownershipUpdater').update(
-        this.get('modelType'),
-        this.get('model.id'),
-        this.get('model.ownedBy')
-      ).then(() => {
-        this.sendAction('closeModal');
-        this.get('toast').success(
-          this.get('successfulSharing'),
-          'Sharing Success',
-          this._toastConfig
-        );
+      this.currentUser.createCompositeGroup(this.selectedUsers).then(() => {
```

Comment:
Be careful, it should not create a CompositeGroup if you're dealing with classic sharing (Company, Teams, Current User)